BrainPortal/app/controllers/nh_messages_controller.rb

```diff
@@ -0,0 +1,117 @@
+#
+# NeuroHub Project
+#
+# Copyright (C) 2020
+# The Royal Institution for the Advancement of Learning
+# McGill University
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+# RESTful controller for managing Messages.
+class NhMessagesController < NeurohubApplicationController
+  Revision_info = CbrainFileRevision[__FILE__] #:nodoc:
+
+  include Pagy::Backend
+
+  before_action :login_required
+
+  # GET /messages
+  # GET /messages.xml
+  def index #:nodoc:
+    @messages        = neurohub_messages
+    @messages_count  = @messages.count
+    @read_count      = @messages.where(:user_id => current_user.id, :read => true).count
+    @unread_count    = @messages.where(:user_id => current_user.id, :read => false).count
+    @page, @per_page = pagination_check(@messages, :nh_messages)
+    @pagy, @messages = pagy(@messages, :items => @per_page)
+  end
+
+  def new #:nodoc:
+    @message        = Message.new # blank object for new() form.
+    @message.header = "A personal message from #{current_user.full_name.presence || current_user.login}"
+    nh_projects     = find_nh_projects(current_user)
+    contacts        = nh_projects.map { |x| x.users }.flatten.uniq.map { |x| x.own_group }
+    @recepients     = (current_user.assignable_groups.order(:name) & nh_projects | contacts).sort_by { |g| g.name }
+  end
+
+  # POST /messages
+  # POST /messages.xml
+  def create #:nodoc:
+    @message              = Message.new(message_params)
+    @message.message_type = :communication
+    @message.sender_id    = current_user.id
+
+    if @message.header.blank?
+      @message.errors.add(:header, "cannot be left blank.")
+    end
+
+    @group_id = params[:group_id]
+    if @group_id.blank?
+      @message.errors.add(:base, "You need to specify the project whose members will receive this message.")
+    elsif @message.errors.empty?
+      group = current_user.assignable_groups.find(@group_id)
```

Comment:
When I try to send a message from a standard user to another standard user, it fails at this line: 

> Couldn't find Group with 'id'=13 [WHERE `groups`.`id` IN (3, 7, 20, 23, 29, 36, 55, 56, 83)]

even if I was able to select this user from the list when sending the message

Reply:
corrected